src/data/literal.rs

```diff
@@ -273,17 +270,7 @@ impl<'a> Literal<'a> {
 
     pub fn like(&self, other: &Literal<'a>) -> Result<Self> {
         match (self, other) {
-            (Text(l), Text(r)) => Ok(Boolean(Regex::new(
-                &format!(
-                    "^{}$",
-                    regex::escape(r)
-                        .replace("%", ".*")
-                        .replace("_", ".")
-                )
-            )
-            .map_err(|e| LiteralError::FailedToParsePattern(e.to_string()))?
-            .is_match(l)
-        )),
+            (Text(l), Text(r)) => Ok(Boolean(l.as_ref().like(&r)?)),
```

Comment:
Tiny comment, you can simply use `l.like(b).map(Boolean)` thanks to smart Rust compiler :)

Reply:
still amazed how amazing rust compiler is :D
anyway thanks for the knowledge